mitodl/micromasters
# Switch to yarn package manager instead of npm

closes #1884 

this pr:

- installs and configures `yarn` in the watch docker container
- checks in a `yarn.lock` file and deletes `npm-shrinkwrap.json`
- sets up the travis docker containers to use yarn

Merging this will require a little bit of follow-up. I changed the `Dockerfile-travis-watch-build` file to install yarn, and then built a new image which is up on docker hub tagged `mm_watch_travis_yarn`. I changed the `Dockerfile-travis-watch` to inherit from this image, so that there's no issue with building this PR before it's been merged. Once the PR is merged and all open PRs are rebased over master we can introduce a commit to undo this change, by pushing a new yarn image to `mm_watch_travis` and then changing `Dockerfile-travis-watch` back to inherit from that.


#### testing

mac:
- install yarn 0.17.9 from npm (`npm install -g yarn@0.17.9`)
- make sure you can install everything and that the app works and so on

linux:
- delete `node_modules`
- pull down the branch and rebuild your `watch` container (`docker-compose build watch`)
- ensure you can boot the app
- ensure you can run some yarn commands (like `yarn install --pure-lockfile`)

More info on yarn CLI usage here: https://yarnpkg.com/en/docs/yarn-workflow
